### ISSUE TITLE
`triton-opt` crashes in the conda environment

### ISSUE THREAD

**Issue by leshikus** [intel/intel-xpu-backend-for-triton#672]
`triton-opt` crashes with the following stack dump:
```
Stack dump:
0.	Program arguments: /runner/_work/intel-xpu-backend-for-triton/intel-xpu-backend-for-triton/python/build/cmake.linux-x86_64-cpython-3.9/bin/triton-opt -convert-tritongen-to-llvm -split-input-file /runner/_work/intel-xpu-backend-for-triton/intel-xpu-backend-for-triton/test/TritonGEN/tritongen-to-llvm.mlir
 #0 0x0000558f48ae27fd llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/runner/_work/intel-xpu-backend-for-triton/intel-xpu-backend-for-triton/python/build/cmake.linux-x86_64-cpython-3.9/bin/triton-opt+0x2c987fd)
 #1 0x0000558f48adfbfb llvm::sys::RunSignalHandlers() (/runner/_work/intel-xpu-backend-for-triton/intel-xpu-backend-for-triton/python/build/cmake.linux-x86_64-cpython-3.9/bin/triton-opt+0x2c95bfb)
 #2 0x0000558f48adfd55 SignalHandler(int) Signals.cpp:0:0
 #3 0x00007fbff0827520 (/lib/x86_64-linux-gnu/libc.so.6+0x42520)
 #4 0x0000558f46d34fe0 mlir::Type::getTypeID() (.isra.0) LLVMTypes.cpp:0:0
 #5 0x0000558f46d44dd0 mlir::LLVM::LLVMFunctionType::verify(llvm::function_ref<mlir::InFlightDiagnostic ()>, mlir::Type, llvm::ArrayRef<mlir::Type>, bool) (.constprop.0) LLVMTypes.cpp:0:0
 #6 0x0000558f46d44ffe mlir::LLVM::LLVMFunctionType::get(mlir::Type, llvm::ArrayRef<mlir::Type>, bool) (/runner/_work/intel-xpu-backend-for-triton/intel-xpu-backend-for-triton/python/build/cmake.linux-x86_64-cpython-3.9/bin/triton-opt+0xefaffe)
 #7 0x0000558f46b2b780 mlir::LLVM::lookupOrCreateFn(mlir::ModuleOp, llvm::StringRef, llvm::ArrayRef<mlir::Type>, mlir::Type, bool) (/runner/_work/intel-xpu-backend-for-triton/intel-xpu-backend-for-triton/python/build/cmake.linux-x86_64-cpython-3.9/bin/triton-opt+0xce[178](https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/8271207886/job/22630446795#step:13:179)0)
 #8 0x0000558f46439d55 createGenISADPAS(mlir::triton::TritonGEN::MatrixDPASOp, mlir::ConversionPatternRewriter&) /runner/_work/intel-xpu-backend-for-triton/intel-xpu-backend-for-triton/third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp:209:18
```

I've added `set -e` to the `test-triton.sh` script. Removing it makes the tests pass, yet the stack trace is still visible in logs,
https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/8270345206/job/22627638904#step:13:163

**Comment by leshikus** [comment]
today I see the following results on the fresh branch https://github.com/intel/intel-xpu-backend-for-triton/tree/lesh/conda

```
Failed Tests (2):
  TRITON :: Conversion/tritongpu_to_gen.mlir
  TRITON :: TritonGEN/tritongen-to-llvm.mlir
```

**Comment by whitneywhtsang** [comment]
Fixed by #703: https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/8351175709/job/22859059250
